dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Relation.java

```diff
@@ -14,11 +15,13 @@
 
 public final class Relation implements Dependent<Relation> {
 
+    private final Wmm wmm;
```

Comment:
Why do we want to store the Wmm associated to a relation?

Reply:
I used this to be able to move configuration data into the `Wmm`.
More precisely, `Acyclicity` had this `reduceAcyclicityEncodeSet` option, which I moved into `Wmm` cause (1) we probably never want to configure each `Acyclicity` axiom separately, and (2) we can log this setting once instead of per axiom.
But now `Acyclicity` needs to know which `Wmm` it belongs to in order to retrieve the configuration data. By attaching the `Wmm` to `Relation`, every `Constraint/Axiom` is able to access the `Wmm`.
It is similar to how `Event` knows which `Function/Program` it belongs to.

While I don't like configuring a data object like `Wmm`, it was the easiest solution for now. To somewhat avoid the issue of configurating the `Wmm` directly, I also added the inner `Config` class which holds the data.